src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder();
+        LocalDate localDateFrom = parseDateFromString(dateFrom);
+        LocalDate localDateTo = parseDateFromString(dateTo);
+
+        report.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+
+        for (String currentName : names) {
+
+            int salaryAmount = 0;
+            for (String record : data) {
+                if (record == null || report.isEmpty()) {
+                    continue;
+                }
+
+                String[] parseRecord = record.split(" ");
+                LocalDate day = parseDateFromString(parseRecord[0]);
+                String name = parseRecord[1];
+                int hours = parseNumberFromString(parseRecord[2]);
+                int daySalary = parseNumberFromString(parseRecord[3]);
+
+                if ((day.equals(localDateFrom) || day.isAfter(localDateFrom))
```

Comment:
This condition is a bit complex and hard to read. Consider simplifying it. You can use `LocalDate.isBefore()` and `LocalDate.isAfter()` methods to check if the `day` is within the range between `localDateFrom` and `localDateTo`.